### PR TITLE
Idva6 790 email producer

### PR DESCRIPTION
**Pull request by krishna-patel-ch**
__Short description outlining key changes/additions__

An email service that can be used to send one of six email types.

__JIRA Ticket Number__

IDVA6-790

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [x] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -13,3 +13,13 @@ management.endpoint.health.show-details=never
 
 internal.api.url=${INTERNAL_API_URL}
 private.api.url=${API_URL}
+
+email.producer.appId=accounts-association-api
```
Comment (kkonuganti-ch) [inline src/main/resources/application.properties:17]: use spring.application.name

**Review by kkonuganti-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/resources/application.properties (hunk 1 above)
Comment (krishna-patel-ch) [reply src/main/resources/application.properties:17]: I think the email producer library requires it to be called email.producer.appId

**Pull request merged**